handler/src/test/java/io/netty/handler/ssl/SslHandlerTest.java

```diff
@@ -873,6 +875,222 @@ public void channelInactive(ChannelHandlerContext ctx) {
+                        @Override
+                        protected void initChannel(Channel ch) throws Exception {
+                            final SslHandler handler = sslServerCtx.newHandler(ch.alloc());
+                            handler.setWrapDataSize(-1);
+                            // The JDK SSLEngineImpl has been observed to write more than 16384 packet size with this
+                            // cipher.
+                            handler.engine().setEnabledCipherSuites(new String[]{"TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA"});
+                            ch.pipeline().addLast(handler);
+                            ch.pipeline().addLast(new ChannelInboundHandlerAdapter() {
+                                private boolean sentData;
+                                private Throwable writeCause;
+
+                                @Override
+                                public void userEventTriggered(ChannelHandlerContext ctx, Object evt) {
+                                    if (evt instanceof SslHandshakeCompletionEvent) {
+                                        SslHandshakeCompletionEvent sslEvt = (SslHandshakeCompletionEvent) evt;
+                                        if (sslEvt.isSuccess()) {
+                                            CompositeByteBuf content = ctx.alloc().compositeDirectBuffer(numComponents);
+                                            for (int i = 0; i < numComponents; ++i) {
+                                                ByteBuf buf = ctx.alloc().directBuffer(singleComponentSize);
+                                                buf.writerIndex(buf.writerIndex() + singleComponentSize);
+                                                content.addComponent(true, buf);
+                                            }
+                                            ctx.writeAndFlush(content).addListener(new ChannelFutureListener() {
+                                                @Override
+                                                public void operationComplete(ChannelFuture future) throws Exception {
+                                                    writeCause = future.cause();
+                                                    if (writeCause == null) {
+                                                        sentData = true;
+                                                    }
+                                                }
+                                            });
+                                        } else {
+                                            donePromise.tryFailure(sslEvt.cause());
+                                        }
+                                    }
+                                    ctx.fireUserEventTriggered(evt);
+                                }
+
+                                @Override
+                                public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
```

Comment:
Consider using AssertionError

Reply:
`AssertionError(String, Throwable)` is jdk 1.7+ and is noisy in the IDE. I'll just stick with `IllegalStateException` here ... end result should be the same.

Reply:
sounds good to me... Thanks for clarify